### PR TITLE
Docs-component-architecture

### PR DESCRIPTION
**Pull request by bmckercher123**
## Description

- Backlog item to add new component architecture diagram, and update topic for OS.
- Update data-path file to include ebpf and edits approved by Alex
- Merge to nightly and cherrypick to /master

### CODE REVIEW

File: reference/architecture/data-path.md [hunk 1]
```diff
@@ -49,31 +48,17 @@ tapa429fb36-04. Other workloads, with the .21, .22 and .23 addresses,
...
-traffic off the data center’s core network, and as a secondary defense
-in case it is possible for a rogue workload to compromise its local
-host.
-
 ## Is that all?
 
 As far as the static data path is concerned, yes. It’s just a
 combination of responding to workload ARP requests with the host MAC, IP
-routing and iptables. There’s a great deal more to Calico in terms of
+routing and iptables or ebpf. There’s a great deal more to {{site.prodname}} in terms of
```
Comment (bcreane) [inline reference/architecture/data-path.md:61]: `eBPF` needs to be camel-cased.

File: reference/architecture/overview.md [hunk 2]
```diff
@@ -1,204 +1,145 @@
...
+   - Simpler to manage because it does not require an extra datastore
+   - Use Kubernetes RBAC to control access to {{site.prodname}} resources
+   - Use Kubernetes audit logging to generate audit logs of changes to {{site.prodname}} resources
 
-{{site.prodname}} deploys a BGP client on every node that also hosts a [Felix](#felix). The role of the BGP client is to read routing state that [Felix](#felix) programs into the kernel and
-distribute it around the data center.
+- **etcd**
 
-The BGP client is responsible for performing the following task:
+   etcd is a consistent, highhly-available distributed key-value store that provides data storage for the {{site.prodname}} network, and forcommunications between components. etcd is supported only for protecting non-cluster hosts (as of {{site.prodname}} v3.1). For completeness, etcd advantages are:
```
Comment (bcreane) [inline reference/architecture/overview.md:107]: typo `highhly`
also suggest putting `etcd`  in back ticks

**Review by bcreane — APPROVED** [review]
looks good. A couple of nits. Please make sure to **squash** this before committing.

File: reference/architecture/overview.md (hunk 2 above)
Comment (bmckercher123) [reply reference/architecture/overview.md]: Fixed typo, added back ticks.

File: reference/architecture/data-path.md (hunk 1 above)
Comment (bmckercher123) [reply reference/architecture/data-path.md]: Fixed.

**Pull request closed without merging**